docs/design-docs/08-package-variant.md

```diff
@@ -292,24 +300,51 @@ The package variant set controller is designed to fill this common need. This
+- a template for generating one PackageVariant per target
 
 Three types of targeting are supported:
-- An explicit list of repository and package names
-- A label selector for Repository objects along with a way to generate package
-  names from those objects
-- An arbitrary object selector, along with ways to generate repository and
-  package names from those objects
-
-*Figure 5* shows an example of creating PackageVariant resources based upon the
-explicitly list of repository and package names.
+- An explicit list of repositories and package names
+- A label selector for Repository objects
+- An arbitrary object selector
 
-![Figure 5: List of Targets](packagevariantset-target-package.png)
+Rules for generating a PackageVariant are associated with a list of targets
+using a template. That template can have explicit values for various
+PackageVariant fields, or it can use [Common Expression Language
+(CEL)](https://github.com/google/cel-go) expressions to specify the field
+values.
 
-Rules for generating PackageVariant resources can also access the fields of
-the target objects, allowing the package context values, injection selectors,
-and function inputs to vary based upon the target.
+*Figure 5* shows an example of creating PackageVariant resources based upon the
+explicitly list of repositories. In this example, for the `cluster-01` and
+`cluster-02` repositories, no template is defined the resulting PackageVariants;
+it simply takes the defaults. However, for `cluster-03`, a template is defined
+to change the downstream package name to `bar`.
+
+| ![Figure 5: PackageVariantSet with Repository List](packagevariantset-target-list.png) |
+| :---: |
+| *Figure 5: PackageVariantSet with Repository List* |
+
+It is also possible to target the same package to a repository more than once,
+using different names. This is useful, for example, if the package is used to
+provision namespaces and you would like to provision many namespaces in the same
+cluster. It is also useful if a repository is shared across multiple clusters.
+In *Figure 6*, two PackageVariant resources for creating the `foo` package in
+the repository `cluster-01` are generated, one for each listed package name.
+Since no `packageNames` field is listed for `cluster-02`, only one instance is
+created for that repository.
+
+| ![Figure 6: PackageVariantSet with Package List](packagevariantset-target-list-with-packages.png) |
+| :---: |
+| *Figure 6: PackageVariantSet with Package List* |
+
+*Figure 7* shows an example that combines a repository label selector with
+configuration injection that various based upon the target. The template for the
+PackageVariant includes a CEL expression for the one of the injectors, so that
+the injection varies systematically based upon attributes of the target.
+
+| ![Figure 7: PackageVariantSet with Repository Selector](packagevariantset-target-repository-selector.png) |
+| :---: |
+| *Figure 7: PackageVariantSet with Repository Selector* |
```

Comment:
Should the template follow the hierarchical structure of the PackageVariant resource spec? So in this case, should we make the template be:
```yaml
...
template:
  downstream:
    package: bar
  injectors:
  - nameExpr: "target.labels['region'] + '-endpoints'"
```